starcheck/src/lib/Ska/Starcheck/Obsid.pm

```diff
@@ -23,14 +23,97 @@ use warnings;
+    :param dither_y: dither_y in arcsecs (guide dither)
+    :param dither_z: dither_z in arcsecs (guide dither)
+    :param yellow_lim: yellow limit centroid offset threshold limit (in arcsecs)
+    :param red_lim: red limit centroid offset threshold limit (in arcsecs)
+    :return imposters: list of dictionaries with keys that define the index, the imposter mag,
+             a 'status' key that has value 0 if the code to get the imposter mag ran successfully,
+             calculated centroid offset, and star or fid info to make a warning.
+    """
+
+    idxs = [int(idx) for idx in idxs]
+    yags = [float(yag) for yag in yags]
+    zags = [float(zag) for zag in zags]
+    mags = [float(mag) for mag in mags]
+    types = [t.decode('ascii') for t in types]
+    t_ccd = float(t_ccd)
+    date = date.decode('ascii')
+    dither_y = float(dither_y)
+    dither_z = float(dither_z)
+
+    dark = aca_dark.get_dark_cal_image(date=date, t_ccd_ref=t_ccd, aca_image=True)
+
+
+    def imposter_offset(cand_mag, imposter_mag):
+        """
+        For a given candidate star and the pseudomagnitude of the brightest 2x2 imposter
+        calculate the max offset of the imposter counts are at the edge of the 6x6
+        (as if they were in one pixel).  This is somewhat the inverse of proseco.get_pixmag_for_offset
+        """
+        cand_counts = mag_to_count_rate(cand_mag)
+        spoil_counts = mag_to_count_rate(imposter_mag)
+        return spoil_counts * 3 * 5 / (spoil_counts + cand_counts)
+
+    imposters = []
+    for idx, yag, zag, mag, ctype in zip(idxs, yags, zags, mags, types):
+        if ctype in ['BOT', 'GUI', 'FID']:
+            if ctype in ['BOT', 'GUI']:
+                dither = ACABox((dither_y, dither_z))
+            else:
+                dither = ACABox((5.0, 5.0))
+            row, col = yagzag_to_pixels(yag, zag, allow_bad=True)
+            # Handle any errors in get_imposter_mags with a try/except.  This doesn't
+            # try to pass back a message.  Most likely this will only fail if the star
+            # or fid is completely off the CCD and will have other warning.
+            try:
+                # get_imposter_mags takes a Table of candidates as its first argument, so construct
+                # a single-candidate table `entries`
+                entries = Table([{'idx': idx, 'row': row, 'col': col, 'mag': mag, 'type': ctype}])
+                imp_mags, imp_rows, imp_cols = get_imposter_mags(entries, dark, dither)
+                offset = imposter_offset(mag, imp_mags[0])
+                imposters.append({'idx': int(idx), 'status': int(0),
+                                  'entry_row': float(row), 'entry_col': float(col),
+                                  'bad2_row': float(imp_rows[0]), 'bad2_col': float(imp_cols[0]),
+                                  'bad2_mag': float(imp_mags[0]), 'offset': float(offset)})
+            except:
+                imposters.append({'idx': int(idx), 'status': int(1)})
```

Comment:
I don't want to necessarily try to skip running this code for a star or fid off the ccd or too close to the edge, so having some error handling seems to make sense.  Whether that handling should just skip the questionable idx altogether, provide a more specific error message, or do what I have here which is to pass back just a failed status, is a separate question.  Thus far, this has only failed for undercovers with fid lights commanded off the ccd in regression weeks.

Reply:
Failed status like you have seems fine.